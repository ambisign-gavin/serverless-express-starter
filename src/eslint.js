// @flow
import execa from 'execa';
import fs from 'fs';
import { join } from 'path';

class Eslint {
    getPackages(): Array<string> {
        return [
            'eslint@^5.0.0',
            'babel-eslint@^10.0.0'
        ];
    }

    async init(projectPath: string): execa.ExecaChildProcess {
        return execa('node_modules/eslint/bin/eslint.js',[
            '--init'
        ], {
            cwd: projectPath,
            stdio: 'inherit'
        });
    }

    rejectParserConfig(projectPath: string) {
        const jsConfig = join(projectPath, '.eslintrc.js');
        if (fs.existsSync(jsConfig)) {
            let eslintrc: { default: Object, parser: string } = require(jsConfig);
            delete eslintrc['default'];
            eslintrc['parser'] = 'babel-eslint';
            const newEslintrc = 'module.exports = ' + JSON.stringify(eslintrc, null, 4) + '\n';
            fs.writeFileSync(jsConfig, newEslintrc);
        }

        const jsonConfig = join(projectPath, '.eslintrc.json');
        if (fs.existsSync(jsonConfig)) {
            let eslintrc = require(jsonConfig);
            eslintrc = {
                ...eslintrc,
                parser: 'babel-eslint'
            };
            const newEslintrc = JSON.stringify(eslintrc, null, 4) + '\n';
            fs.writeFileSync(jsonConfig, newEslintrc);
        }
    }
}

const eslint = new Eslint();

export default eslint;