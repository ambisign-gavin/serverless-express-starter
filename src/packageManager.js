// @flow
import execa from 'execa';
import { type PackageManagerType } from './type';

const dependenciesModule = [
    'express@^4.0.0',
    'serverless-http@^1.0.0',
];

const devDependenciesModule = [
    '@babel/cli@^7.0.0',
    '@babel/core@^7.0.0',
    '@babel/plugin-transform-modules-commonjs@^7.0.0',
    'cli-confirm@^1.0.0',
    'serverless-offline@^4.0.0',
];

class PackageManager {
    async install(packageManager: PackageManagerType, projectPath: string): Promise<void> {
        try {
            await execa(packageManager,[
                'add',
                ...dependenciesModule
            ], {
                cwd: projectPath,
                stdio: 'inherit'
            });
        
            await execa(packageManager,[
                'add',
                '-D',
                ...devDependenciesModule
            ], {
                cwd: projectPath,
                stdio: 'inherit'
            });
        } catch (error) {
            return Promise.reject(error);
        }
        return Promise.resolve();
    }

    generatePackageJson(packageManager: PackageManagerType, name: string, description: string): Object {
        const packageJson = {
            name,
            main: 'index.js',
            version: '1.0.0',
            description,
            scripts: {
                'build': 'babel src/ -d lib/',
                'local': `${packageManager} run build && sls offline start`,
                'deploy:dev': `${packageManager} run build && serverless deploy --stage dev`,
                'confirm:production': 'cli-confirm \' Do you want to deploy production server? \'',
                'deploy:production': `${packageManager} run confirm:production && ${packageManager} run build && serverless deploy --stage production`
            },
            dependencies: {},
            devDependencies: {},
            keywords: [],
        };
        return packageJson;
    }

}

const packageInstaller = new PackageManager();

export default packageInstaller;