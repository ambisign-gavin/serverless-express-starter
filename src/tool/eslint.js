// @flow
import execa from 'execa';
import fs from 'fs';
import { join } from 'path';
import YAML from 'yaml';

class Eslint {
    getInstallPackages(): Array<string> {
        return [
            'eslint@^5.0.0',
            'babel-eslint@^10.0.0'
        ];
    }

    init(projectPath: string): execa.ExecaChildProcess {
        return execa('node_modules/eslint/bin/eslint.js',[
            '--init'
        ], {
            cwd: projectPath,
            stdio: 'inherit'
        });
    }

    loadConfig(configPath: string): Object {
        let config = {};
        const jsConfig = join(configPath, '.eslintrc.js');
        if (fs.existsSync(jsConfig)) {
            config = require(jsConfig);
        }

        const jsonConfig = join(configPath, '.eslintrc.json');
        if (fs.existsSync(jsonConfig)) {
            config = require(jsonConfig);
        }

        const ymlConfig = join(configPath, '.eslintrc.yml');
        if (fs.existsSync(ymlConfig)) {
            const file = fs.readFileSync(ymlConfig, 'utf8');
            config = YAML.parse(file);
        }
        return config;
    }

    saveConfig(configPath: string, config: Object) {
        const jsConfig = join(configPath, '.eslintrc.js');
        if (fs.existsSync(jsConfig)) {
            const newEslintrc = 'module.exports = ' + JSON.stringify(config, null, 4) + '\n';
            fs.writeFileSync(jsConfig, newEslintrc);
        }

        const jsonConfig = join(configPath, '.eslintrc.json');
        if (fs.existsSync(jsonConfig)) {
            const newEslintrc = JSON.stringify(config, null, 4) + '\n';
            fs.writeFileSync(jsonConfig, newEslintrc);
        }

        const ymlConfig = join(configPath, '.eslintrc.yml');
        if (fs.existsSync(ymlConfig)) {
            const newEslintrc = YAML.stringify(config);
            fs.writeFileSync(ymlConfig, newEslintrc);
        }
    }

    fixScripts(projectPath: string): execa.ExecaChildProcess {
        return execa('node_modules/eslint/bin/eslint.js',[
            '--fix',
            './src/*.js'
        ], {
            cwd: projectPath,
            stdio: 'inherit'
        });
    }

}

const eslint = new Eslint();

export default eslint;