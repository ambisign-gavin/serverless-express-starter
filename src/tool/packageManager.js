// @flow
import execa from 'execa';
import { type PackageManagerPlatform } from '../type';

class PackageManager {
    async install(packageManager: PackageManagerPlatform, projectPath: string, packages: Array<string>, isDev: boolean = false): Promise<void> {
        try {
            const commands = [
                ...packages
            ];
            if (isDev) {
                commands.unshift('-D');
            }
            await execa(packageManager,[
                'add',
                ...commands
            ], {
                cwd: projectPath,
                stdio: 'inherit'
            });
        } catch (error) {
            return Promise.reject(error);
        }
        return Promise.resolve();
    }
    generatePackageJson(packageManager: PackageManagerPlatform, name: string, description: string): Object {
        const packageJson = {
            name,
            main: 'index.js',
            version: '1.0.0',
            description,
            scripts: {
                'build': 'babel src/ -d lib/',
                'local': `${packageManager} run build && sls offline start`,
                'deploy:dev': `${packageManager} run build && serverless deploy --stage dev`,
                'confirm:production': 'cli-confirm \' Do you want to deploy production server? \'',
                'deploy:production': `${packageManager} run confirm:production && ${packageManager} run build && serverless deploy --stage production`
            },
            dependencies: {},
            devDependencies: {},
            keywords: [],
        };
        return packageJson;
    }

}

const packageManager = new PackageManager();

export default packageManager;