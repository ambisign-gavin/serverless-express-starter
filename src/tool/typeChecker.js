// @flow
import { type TypeCheckerPlatform } from '../type';
import execa from 'execa';

class TypeChecker {
    generateTemplatePath(typeCheckerPlatform: TypeCheckerPlatform): string {
        switch (typeCheckerPlatform) {
        case 'none':
            return 'template/default/';
        case 'flow':
            return 'template/flow/';
        default:
            return 'template/default/';
        }
    }

    getInstallPackages(typeCheckerPlatform: TypeCheckerPlatform): Array<string> {
        if (typeCheckerPlatform === 'flow') {
            return [
                'flow-bin@^0.95.0',
                'flow-typed@^2.0.0',
                '@babel/preset-flow'
            ];
        }
        return [];
    }
    
    runExtraSettings(typeCheckerPlatform: TypeCheckerPlatform, projectPath: string): execa.ExecaChildProcess {
        if (typeCheckerPlatform === 'flow') {
            return execa('flow-typed', [
                'install',
                'express@^4.0.0'
            ], {
                cwd: projectPath
            });
        }
    }
}

const typeChecker = new TypeChecker();

export default typeChecker;