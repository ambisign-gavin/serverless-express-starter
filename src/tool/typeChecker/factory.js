// @flow
import { type TypeCheckerPlatform } from '../../type';
import { TypeChecker } from './interface';
import { FlowTypeChecker } from './flow';
import { NoneTypeChecker } from './noneChecker';

export const typeCheckerFactory = {
    getTypeChecker(typeCheckerPlatform: TypeCheckerPlatform): TypeChecker {
        switch (typeCheckerPlatform) {
        case 'flow':
            return new FlowTypeChecker();
        default:
            return new NoneTypeChecker();
        }
    }
};