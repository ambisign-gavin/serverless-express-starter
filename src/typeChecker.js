// @flow
import { type TypeCheckerPlatform } from './type';
import execa from 'execa';

class TypeChecker {
    generateTemplatePath(typeCheckerPlatform: TypeCheckerPlatform): string {
        switch (typeCheckerPlatform) {
        case 'none':
            return 'template/default/';
        case 'flow':
            return 'template/flow/';
        default:
            return 'template/default/';
        }
    }

    getInstallPackage(typeCheckerPlatform: TypeCheckerPlatform): Array<string> {
        if (typeCheckerPlatform === 'flow') {
            return [
                'flow-bin@^0.95.0',
                'flow-typed@^2.0.0',
                '@babel/preset-flow'
            ];
        }
        return [];
    }
    
    async runExtraSettings(typeCheckerPlatform: TypeCheckerPlatform, projectPath: string) {
        if (typeCheckerPlatform === 'flow') {
            await execa('flow-typed', [
                'install',
                'express@^4.0.0'
            ], {
                cwd: projectPath
            });
        }
    }
}

const typeChecker = new TypeChecker();

export default typeChecker;