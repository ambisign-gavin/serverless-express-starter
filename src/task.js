// @flow
import chalk from 'chalk';
import { writeFileSync } from 'fs';
import del from 'del';
import { join } from 'path';
import execa from 'execa';
import { InquirerRobot } from './tool/inquirer';
import packageManager from './tool/packageManager';
import eslint from './tool/eslint';

class Task {
    async createFiles(inquirerRobot: InquirerRobot) {
        try {
            const projectName = inquirerRobot.name;
            const projectPath = join(process.cwd(), inquirerRobot.name);
            const pkgManagerPlatform = inquirerRobot.pkgManager;
            const typeChecker = inquirerRobot.typeChecker;
    
            await execa('git', [
                'clone',
                'https://github.com/ambisign-gavin/serverless-express-starter.git',
                projectName
            ]);
    
            await del([
                join(projectPath, 'src'),
                join(projectPath, 'bin'),
                join(projectPath, '.gitignore'),
                join(projectPath, '.eslintrc.js'),
                join(projectPath, '.flowconfig'),
                join(projectPath, '.babelrc'),
                join(projectPath, '.git'),
                join(projectPath, 'package-lock.json'),
            ]);
    
            execa.shellSync(`cp -a ${typeChecker.generateTemplatePath()}. ./`, {
                cwd: projectPath
            });
    
            await del([
                join(projectPath, 'template'),
            ]);
    
            const packageContext = packageManager.generatePackageJson(
                pkgManagerPlatform,
                projectName,
                inquirerRobot.description
            );
    
            writeFileSync(join(projectPath, 'package.json'), JSON.stringify(packageContext, null, 2) + '\n');
        } catch (error) {
            console.log('An error occurred while creating files:', error);
        }
        
    }
    
    async installPackages(inquirerRobot: InquirerRobot) {
        try {
            let dependenciesModule = [
                'express@^4.0.0',
                'serverless-http@^1.0.0',
            ];
            
            let devDependenciesModule = [
                '@babel/cli@^7.0.0',
                '@babel/core@^7.0.0',
                '@babel/plugin-transform-modules-commonjs@^7.0.0',
                'cli-confirm@^1.0.0',
                'serverless-offline@^4.0.0',
            ];
    
            const projectPath = join(process.cwd(), inquirerRobot.name);
            const pkgManagerPlatform = inquirerRobot.pkgManager;
            const typeChecker = inquirerRobot.typeChecker;
    
            if (typeChecker.getInstallPackages().length > 0) {
                devDependenciesModule.push(...typeChecker.getInstallPackages());
            }
            
            if (inquirerRobot.isUsedEslint) {
                devDependenciesModule.push(...eslint.getInstallPackages());
            }
    
            await packageManager.install(
                pkgManagerPlatform,
                projectPath,
                dependenciesModule,
                false,
            );
    
            await packageManager.install(
                pkgManagerPlatform,
                projectPath,
                devDependenciesModule,
                true,
            );
        } catch (error) {
            console.log('An error occurred while installing packages:', error);
        }
        
    }
    
    async runExtraSettings(inquirerRobot: InquirerRobot) {
        try {
            const projectPath = join(process.cwd(), inquirerRobot.name);
            const typeChecker = inquirerRobot.typeChecker;

            await typeChecker.runExtraSettings(projectPath);

            if (inquirerRobot.isUsedEslint) {
                await eslint.init(projectPath);
                eslint.injectionParserConfig(projectPath);
                await eslint.fixScripts(projectPath);
            }
        } catch (error) {
            console.log('An error occurred while setting plugins:', error);
        }
        
    }
    
    showCompleteMessages() {
        const messages = `
It's all done! You can start to develop your mighty project!

You can use:

  npm/yarn run local               Run the local server.
  npm/yarn run deploy:dev          Deploy the server of dev stage.
  npm/yarn run deploy:production   Deploy the server of production stage.

Enjoy it!
        `;
        console.log(chalk.green(messages));
    }
}

const task = new Task();

export default task;