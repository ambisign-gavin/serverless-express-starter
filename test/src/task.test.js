// @flow
import execa from 'execa';
import task from '../../src/task';
import inquirerRobot from '../../src/tool/inquirer';
import del from 'del';
import fs from 'fs';
import typeChecker from '../../src/tool/typeChecker';
import packageManager from '../../src/tool/packageManager';
import eslint from '../../src/tool/eslint';

jest.mock('execa');
jest.mock('del');
jest.mock('fs');
jest.mock('chalk', () => ({
    green: jest.fn().mockImplementation(text => text)
}));
jest.mock('../../src/tool/eslint');
jest.mock('../../src/tool/typeChecker');
jest.mock('../../src/tool/packageManager');

function spyInquirerRobotDefaultValue() {
    jest.spyOn(inquirerRobot, 'name', 'get').mockReturnValue('my-server');
    jest.spyOn(inquirerRobot, 'pkgManager', 'get').mockReturnValue('npm');
    jest.spyOn(inquirerRobot, 'typeChecker', 'get').mockReturnValue('none');
    jest.spyOn(inquirerRobot, 'isUsedEslint', 'get').mockReturnValue(false);
}

function spyProcessCwd() {
    jest.spyOn(process, 'cwd').mockReturnValue('/test/');
}

describe('Create files task', () => {

    beforeAll(() => {
        spyProcessCwd();
        typeChecker.generateTemplatePath.mockReturnValue('/template/test/');
        packageManager.generatePackageJson.mockReturnValue({
            name: 'my-server',
            main: 'index.js',
            version: '1.0.0',
            description: 'This is my server.',
            scripts: {
                'build': 'babel src/ -d lib/',
            },
            dependencies: {},
            devDependencies: {},
            keywords: [],
        });
    });

    beforeEach(async () => {
        jest.clearAllMocks();
        spyInquirerRobotDefaultValue();
        await task.createFiles(inquirerRobot);
    });

    it('should clone git correct', () => {
        expect(execa.mock.calls.length).toEqual(1);
        expect(execa.mock.calls[0][0]).toEqual('git');
        expect(execa.mock.calls[0][1]).toEqual([
            'clone',
            'https://github.com/ambisign-gavin/serverless-express-starter.git',
            'my-server'
        ]);
    });

    it('should delete files correct', () => {
        expect(del.mock.calls.length).toEqual(2);
        expect(del.mock.calls[0][0]).toEqual([
            '/test/my-server/src',
            '/test/my-server/bin',
            '/test/my-server/.gitignore',
            '/test/my-server/.eslintrc.js',
            '/test/my-server/.flowconfig',
            '/test/my-server/.babelrc',
            '/test/my-server/.git',
            '/test/my-server/package-lock.json',
        ]);
        expect(del.mock.calls[1][0]).toEqual([
            '/test/my-server/template',
        ]);
    });

    it('should copy file correct', () => {
        expect(execa.shellSync.mock.calls.length).toEqual(1);
        expect(execa.shellSync.mock.calls[0][0]).toEqual('cp -a /template/test/. ./');
        expect(execa.shellSync.mock.calls[0][1]).toEqual({
            cwd: '/test/my-server'
        });
    });

    it('should write package.json correct', () => {
        expect(fs.writeFileSync.mock.calls.length).toEqual(1);
        expect(fs.writeFileSync.mock.calls[0][0]).toEqual('/test/my-server/package.json');
        expect(fs.writeFileSync.mock.calls[0][1]).toMatchSnapshot();
    });

});

describe('Install packages task', () => {

    beforeAll(() => {
        spyProcessCwd();
        typeChecker.getInstallPackages.mockReturnValue([
            'flow-bin@^0.95.0'
        ]);
        eslint.getInstallPackages.mockReturnValue([
            'eslint@^5.0.0'
        ]);
    });

    beforeEach(() => {
        jest.clearAllMocks();
        spyInquirerRobotDefaultValue();
    });

    it('should install correct with no typeChecker', async () => {
        jest.spyOn(inquirerRobot, 'typeChecker', 'get').mockReturnValue('none');
        await task.installPackages(inquirerRobot);
        expect(packageManager.install.mock.calls.length).toEqual(2);
        expect(packageManager.install.mock.calls[0][0]).toEqual('npm');
        expect(packageManager.install.mock.calls[0][1]).toEqual('/test/my-server');
        expect(packageManager.install.mock.calls[0][2]).toEqual([
            'express@^4.0.0',
            'serverless-http@^1.0.0'
        ]);
        expect(packageManager.install.mock.calls[0][3]).toBeFalsy();

        expect(packageManager.install.mock.calls[1][0]).toEqual('npm');
        expect(packageManager.install.mock.calls[1][1]).toEqual('/test/my-server');
        expect(packageManager.install.mock.calls[1][2]).toEqual([
            '@babel/cli@^7.0.0',
            '@babel/core@^7.0.0',
            '@babel/plugin-transform-modules-commonjs@^7.0.0',
            'cli-confirm@^1.0.0',
            'serverless-offline@^4.0.0',
        ]);
        expect(packageManager.install.mock.calls[1][3]).toBeTruthy();
    });

    it('should install correct with typeChecker', async () => {
        jest.spyOn(inquirerRobot, 'typeChecker', 'get').mockReturnValue('flow');
        await task.installPackages(inquirerRobot);
        expect(packageManager.install.mock.calls.length).toEqual(2);
        expect(packageManager.install.mock.calls[0][0]).toEqual('npm');
        expect(packageManager.install.mock.calls[0][1]).toEqual('/test/my-server');
        expect(packageManager.install.mock.calls[0][2]).toEqual([
            'express@^4.0.0',
            'serverless-http@^1.0.0'
        ]);
        expect(packageManager.install.mock.calls[0][3]).toBeFalsy();

        expect(packageManager.install.mock.calls[1][0]).toEqual('npm');
        expect(packageManager.install.mock.calls[1][1]).toEqual('/test/my-server');
        expect(packageManager.install.mock.calls[1][2]).toEqual([
            '@babel/cli@^7.0.0',
            '@babel/core@^7.0.0',
            '@babel/plugin-transform-modules-commonjs@^7.0.0',
            'cli-confirm@^1.0.0',
            'serverless-offline@^4.0.0',
            'flow-bin@^0.95.0'
        ]);
        expect(packageManager.install.mock.calls[1][3]).toBeTruthy();
    });

    it('should install correct with no eslint', async () => {
        jest.spyOn(inquirerRobot, 'isUsedEslint', 'get').mockReturnValue(false);
        await task.installPackages(inquirerRobot);
        expect(packageManager.install.mock.calls.length).toEqual(2);
        expect(packageManager.install.mock.calls[0][0]).toEqual('npm');
        expect(packageManager.install.mock.calls[0][1]).toEqual('/test/my-server');
        expect(packageManager.install.mock.calls[0][2]).toEqual([
            'express@^4.0.0',
            'serverless-http@^1.0.0'
        ]);
        expect(packageManager.install.mock.calls[0][3]).toBeFalsy();

        expect(packageManager.install.mock.calls[1][0]).toEqual('npm');
        expect(packageManager.install.mock.calls[1][1]).toEqual('/test/my-server');
        expect(packageManager.install.mock.calls[1][2]).toEqual([
            '@babel/cli@^7.0.0',
            '@babel/core@^7.0.0',
            '@babel/plugin-transform-modules-commonjs@^7.0.0',
            'cli-confirm@^1.0.0',
            'serverless-offline@^4.0.0',
        ]);
        expect(packageManager.install.mock.calls[1][3]).toBeTruthy();
    });

    it('should install correct with typeChecker', async () => {
        jest.spyOn(inquirerRobot, 'isUsedEslint', 'get').mockReturnValue(true);
        await task.installPackages(inquirerRobot);
        expect(packageManager.install.mock.calls.length).toEqual(2);
        expect(packageManager.install.mock.calls[0][0]).toEqual('npm');
        expect(packageManager.install.mock.calls[0][1]).toEqual('/test/my-server');
        expect(packageManager.install.mock.calls[0][2]).toEqual([
            'express@^4.0.0',
            'serverless-http@^1.0.0'
        ]);
        expect(packageManager.install.mock.calls[0][3]).toBeFalsy();

        expect(packageManager.install.mock.calls[1][0]).toEqual('npm');
        expect(packageManager.install.mock.calls[1][1]).toEqual('/test/my-server');
        expect(packageManager.install.mock.calls[1][2]).toEqual([
            '@babel/cli@^7.0.0',
            '@babel/core@^7.0.0',
            '@babel/plugin-transform-modules-commonjs@^7.0.0',
            'cli-confirm@^1.0.0',
            'serverless-offline@^4.0.0',
            'eslint@^5.0.0'
        ]);
        expect(packageManager.install.mock.calls[1][3]).toBeTruthy();
    });

});

describe('Setting task', () => {

    beforeAll(() => {
        spyProcessCwd();
    });

    beforeEach(() => {
        jest.clearAllMocks();
        spyInquirerRobotDefaultValue();
    });

    it('should setting correct with type checker', async () => {
        jest.spyOn(inquirerRobot, 'typeChecker', 'get').mockReturnValue('flow');
        await task.runExtraSettings(inquirerRobot);
        expect(typeChecker.runExtraSettings.mock.calls.length).toEqual(1);
    });

    it('should setting correct with eslint', async () => {
        jest.spyOn(inquirerRobot, 'isUsedEslint', 'get').mockReturnValue(true);
        await task.runExtraSettings(inquirerRobot);
        expect(eslint.init.mock.calls.length).toEqual(1);
        expect(eslint.injectionParserConfig.mock.calls.length).toEqual(1);
        expect(eslint.fixScripts.mock.calls.length).toEqual(1);
    });

});

describe('Show complete message of task', () => {
    it('should showed correct', () => {
        let log = jest.spyOn(console, 'log');
        task.showCompleteMessages();
        expect(log.mock.calls.length).toEqual(1);
        expect(log.mock.calls[0][0]).toMatchSnapshot();
    });
});
