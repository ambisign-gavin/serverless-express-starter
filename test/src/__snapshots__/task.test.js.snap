// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create files task should write package.json correct 1`] = `
"{
  \\"name\\": \\"my-server\\",
  \\"main\\": \\"index.js\\",
  \\"version\\": \\"1.0.0\\",
  \\"description\\": \\"This is my server.\\",
  \\"scripts\\": {
    \\"test\\": \\"echo test\\",
    \\"build\\": \\"babel src/ -d lib/\\"
  },
  \\"dependencies\\": {},
  \\"devDependencies\\": {},
  \\"keywords\\": []
}
"
`;

exports[`Show complete message of task should showed correct 1`] = `
"
It's all done! You can start to develop your mighty project!

You can use:

  npm/yarn run local               Run the local server.
  npm/yarn run deploy:dev          Deploy the server of dev stage.
  npm/yarn run deploy:production   Deploy the server of production stage.

Enjoy it!
        "
`;
